apps/system/js/bluetooth.js

```diff
@@ -216,6 +231,70 @@ var Bluetooth = {
   },
 
   /**
+   * Get adapter from bluetooth through promise interface.
+   * XXX: Since BTv1 get onenabled event before onadapteradded event,
+   * We need watch if adapter is retrieved.
+   *
+   * _getAdapterPromise is used to make sure we return the same
+   * promise to outter caller.
+   * resolve is cached in _resolveAdapterPromise and will execute
+   * when the adapter is set.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  adapter: function bt__adapter() {
+    if (!this._getAdapterPromise) {
+      this._getAdapterPromise = new Promise((resolve) => {
+        if (this.defaultAdapter !== null) {
+          resolve(this.defaultAdapter);
+        }
+        // cache the resolve and execute when adapter is ready
+        this._resolveAdapterPromise = resolve;
```

Comment:
If you `resolve()` the promise in case of adapter existed, why you still need to cache the `resolve` function here. You should cache the `resolve` in case of no adapter.